Automattic/amp-wp
# Setup wizard: Container UI and fetching/updating of options

## Summary

This resolves #4700. It makes the following updates:

1. Generates the template for the wizard screen by overriding the WP admin via a filter.
2. Makes plugin options available and editable over REST via the core settings endpoint.
3. Creates the main layout or the React application, including much of the styling and the functionality to navigate between the pages.
4. Creates a React context provider component for interacting with the plugins options over REST. 
5. Creates the component that interrupts the user upon trying to leave with unsaved changes.
6. Creates stubs for the seven pages currently outlined in the designs.

### QA steps

1. Inside your local copy of the repo, fetch the latest updates: `git fetch`
2. Clear any uncommitted changes: `git checkout -- .`
3. Check out this branch with `git checkout feature/4700-setup-onboarding`
4. Rebuild: `composer install && npm install && npm run build`
5. Start your environment: `./bin/local-env/start.sh`
6. Navigate to the wizard page: http://localhost:8890/wp-admin/admin.php?page=amp-setup&amp-new-onboarding=1
7. Verify you can click to the last screen and back.
8. Verify everything matches the designs.

Attn: @jwold 😃 

## Checklist
